src/datasets/utils/filelock.py

```diff
@@ -96,6 +104,24 @@ def __str__(self):
         return temp
 
 
+# Helpers
+# ------------------------------------------------
+
+
+MAX_LOCK_FILENAME_LENGTH = 255
+
+
+def hash_filename_if_too_long(path: str, max_length=MAX_LOCK_FILENAME_LENGTH) -> str:
```

Comment:
As this function is only used by FileLock, maybe it could become a method of that class: it is the class responsibility to check whether the filename is too long and hash it.

Reply:
And I would recommend adding some unit tests for this function (even if they seem quite trivial).